template/template.go

```diff
@@ -170,10 +170,35 @@ func (t *Template) Execute(w io.Writer, input map[string]any) error {
 		return errors.New("execute template: AST is nil")
 	}
 
-	executer := &executer{template: t, w: w, variables: make(map[string]any, len(input))}
+	executer := &executer{template: t, w: w, variables: make(map[string]*ResolvedValue, len(input))}
 
 	for k, v := range input {
-		executer.variables[k] = v
+		var function string
+
+		def := NewResolvedValue(v, WithFormat(func() string { return defaultFormat(v) }))
+
+		switch v.(type) {
+		default:
+			executer.variables[k] = def
+			continue
+		case string:
+			function = "string"
+		case float64, int:
+			function = "number"
+		}
+
+		f, ok := t.registry[function]
```

Comment:
Why look up a function every time we iterate through _input_ range? we can do it before _for_ loop

Reply:
Each input value can be a different type. Do you suggest iterating through the input slice to gather types and then again to create ResolvedValues?

Reply:
We can access registry functions directly without looking them up.

Reply:
Using registry function directly.